enumerable.rb

```diff
@@ -0,0 +1,102 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    (0...length).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_selection = []
+    my_each do |x|
+      new_selection.push(x) if yield(x)
+    end
+    new_selection
+  end
+
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
+
+    new_selection = []
+    my_each do |n|
+      new_selection.push(n)
+      return false unless yield(n)
+    end
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    arr = []
+    my_each do |x|
+      arr.push(x)
+      return true if yield(x)
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    new_selection = []
+    my_each do |x|
+      new_selection.push(x)
+      return true if yield(x)
+    end
+    false
+  end
+
+  def my_count
+    return to_enum(:my_count) unless block_given?
+
+    total = 0
+    (0...length).each do |x|
+      total += 1 if yield(self[x])
+    end
+    total
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ] my_count returns the number of items in enum through enumeration
```ruby
[5, 4, 1].my_count == [5, 4, 1].count => 3
```
- Yours is retrieving the enumerable
![image](https://user-images.githubusercontent.com/52209067/78922946-a3abac80-7a54-11ea-9b01-6ba2275d78b3.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696292-c200ab00-6f72-11ea-8e97-5a2b79256cbc.png)

- [ ] my_count counts the number of items in enum that are equal to the item if an argument is given
```ruby
[5, 4, 1].my_count(4) == [5, 4, 1].count(4) => 1
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78923693-dd30e780-7a55-11ea-9774-59ea7ea9cc9d.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696378-e2306a00-6f72-11ea-9310-0bb30ab9c94d.png)